src/sentry/integrations/messaging.py

```diff
@@ -280,72 +288,122 @@ def expired_link_template(self) -> str:
+
+    def get(self, request: Request, *args, **kwargs) -> HttpResponse:
+        params = kwargs["params"]
+        organization, integration = params["organization"], params["integration"]
+
+        return render_to_response(
+            self.confirmation_template,
+            request=request,
+            context={"organization": organization, "provider": integration.get_provider()},
+        )
 
-        if request.method != "POST":
-            context = {
-                "organization": organization,
-                "provider": integration.get_provider() if integration else None,
-            }
-            return render_to_response(self.confirmation_template, request=request, context=context)
-
-        response = self.execute(idp, params[self.external_id_parameter], request)
-        if response is not None:
-            return response
-
-        self.notify_on_success(integration, params)
-
-        if self.success_metric:
-            provider_slug = self.parent_messaging_spec.provider_slug
-            analytics.record(
-                self.success_metric,
-                provider=provider_slug,
-                actor_id=request.user.id,
-                actor_type=ActorType.USER,
+    def post(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
+        if isinstance(request.user, AnonymousUser):
+            return HttpResponse(status=401)
+
+        try:
+            organization: RpcOrganization = kwargs["organization"]
+            integration: Integration = kwargs["integration"]
+            idp: IdentityProvider = kwargs["idp"]
+
+            params_dict: Mapping[str, Any] = kwargs["params"]
+            external_id: str = params_dict[self.external_id_parameter]
+        except KeyError as e:
+            key = self.get_metric_key("failure.post.missing_params")
+            logger.exception(key)
+            metrics.incr(key, tags={"error": str(e)}, sample_rate=1.0)
+            return self._render_error_page(
+                request,
+                status=400,
+                body_text="HTTP 400: Missing required parameters.",
             )
 
-        team_id = params.get("team_id")
-        context = {"team_id": team_id} if team_id else {}
-        return render_to_response(self.success_template, request=request, context=context)
+        exc_response = self.persist_identity(idp, external_id, request)
+        if exc_response is not None:
+            return exc_response
+
+        self.notify_on_success(integration, external_id, params_dict)
+
+        controller = NotificationController(
+            recipients=[request.user],
+            organization_id=organization.id,
+            provider=self.external_provider_enum,
+        )
+        has_provider_settings = controller.user_has_any_provider_settings(
+            self.external_provider_enum
+        )
+
+        if not has_provider_settings:
+            IntegrationNudgeNotification(organization, request.user, self.provider).send()
```

Comment:
@RyanSkonnord Do we need this nudge still? I think the nudge was to get people to enable Slack back when Slack was disabled by default. But now Slack is enabled by default.

Reply:
I don't know, but I'll check. This is only here because I moved it from [`slack/views/link_identity.py` lines 155-158](https://github.com/getsentry/sentry/pull/75899/files/bd31b63b14da8b82c9994465a71c9b56ad8803c6#diff-2e2302a8e883ab01eb7334c5e15f596cdcb311e289c89aee74b6444d60e4571bL155-L158).

Reply:
I'm pretty sure this is dead code. I kludged out some type-checking errors (and explained in a comment) in https://github.com/getsentry/sentry/pull/75899/commits/f9e21028d9806f20511469416a4537e1ec7d3d9a. I'll open a PR after this one to delete it.